app/models/experiment.rb

```diff
@@ -221,6 +221,12 @@ def update_canvas(
     true
   end
 
+  def update_workflow_img
+    Timeout.timeout(5) { generate_workflow_img }
+  rescue StandardError
+    delay.generate_workflow_img
```

Comment:
Should we log timeout somewhere?

Reply:
I will add some error loging, I think we need to catch any errors, not only timeout.

Reply:
I agree, but we should know what we catch :)